blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -59,64 +59,64 @@ All join provided by Kafka Streams were designed thinking with aggregation in mi
 - (A+B, null)
 - (A+C, null) <== we will never have (A+B+C)
 ```
 
-In our case, we are building a state machine, and we want to keep the last state of execution, meaning we don't want to see all intermediate states (like in aggregation). In that case, we have no other choice but to build [a custom joiner](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java) since Kafka Stream doesn't have any one built-in.
+In our case, we are building a state machine, and want to keep the last state of execution. This means we don't want to see all intermediate states (like in aggregation). In that case, we have no choice but to build [a custom joiner](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java), since Kafka Stream doesn't have a built-in one.
 
-To handle this case, we need to :
-- create a [manually a store](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorFromExecutionTransformer.java) that will save the last state of an execution
-- create a custom [merge function](https://github.com/kestra-io/kestra/blob/8a17519b5b41e8ba0ab8e0044c9c6b1e887ccd94/runner-kafka/src/main/java/io/kestra/runner/kafka/executors/ExecutorMain.java#L216-L246) that will merge the execution stream with the task results stream
-- [get the last value](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java) from the state, add the task result and emit the new state that will be saved finally on the store and the final topics.
+To handle this case, we need to:
+- [Manually create a store](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorFromExecutionTransformer.java) that will save the last state of an execution.
+- Create a custom [merge function](https://github.com/kestra-io/kestra/blob/8a17519b5b41e8ba0ab8e0044c9c6b1e887ccd94/runner-kafka/src/main/java/io/kestra/runner/kafka/executors/ExecutorMain.java#L216-L246) that will merge the execution stream with the task results stream.
+- [Get the last value](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java) from the state, add the task result and produce the new state that will be saved finally on the store and the final topics.
 
-With all of that, we are sure that the execution state will always be the last version whatever the number of tasks results coming in parallel.
+With all the tasks above, we are sure that the execution state will always be the last version, whatever the number of tasks results coming in parallel.
 
-## Easy distributed workload between multiple backends
+## Easy distribution of workload between multiple backends
```

Comment:
We need a sentence or two here providing context to the sub-header. For instance, we could explain how "easy distribution of workload between multiple backends is related to our topic, what it is, and why it is important for this type of task.